### ISSUE TITLE
IAuthenticationResult and IdToken not Serializable

### ISSUE THREAD

**Issue by dsmith270** [AzureAD/microsoft-authentication-library-for-java#498]
Currently there is a serious defect as the classes representing the User Principal are not serializable. The outcome of the successful authentication is stored in the HttpSession. Often HttpSession need to be replicated as with using the standard J2EE distributable tag. Currently this is not working. 

It's easy to demonstrate this defect. Both IAuthenticationResult and IdToken should be marked as Serializable and they are not currently.

```
public boolean serializeUserPrincipal() {
		
		IAuthenticationResult result = (IAuthenticationResult) session.getAttribute(Constants.PRINCIPAL_SESSION_NAME);
		if (result == null) {
			throw new RuntimeException("please log in first");
		}
		
		ByteArrayOutputStream bout=new ByteArrayOutputStream();    
		ObjectOutputStream out;
		try {
			out = new ObjectOutputStream(bout);
			out.writeObject(result);    
			out.flush();   
		} catch (IOException e) {
			e.printStackTrace();
			return false;
		}    
		 
		byte [] bytes = bout.toByteArray();
		ByteArrayInputStream bin = new ByteArrayInputStream(bytes);
		
		IAuthenticationResult output;
		try {
			ObjectInputStream instr = new ObjectInputStream(bin);
			output = (IAuthenticationResult) instr.readObject();
		} catch (IOException e) {
			e.printStackTrace();
			return false;
		} catch (ClassNotFoundException e) {
			e.printStackTrace();
			return false;
		}
		
		System.err.println(output);
		return true;
		
	}
```

This code will fail. 

I've raised a pull request on this. Would appreciate prompt action as it is blocking our AAD migration.

**Comment by siddhijain** [comment]
Thanks for raising the PR and Sorry for the delay on this. Making these classes Serializable looks good to me. I have approved the PR and merge it in. We are planning a msal release tomorrow or early next week. 
Please close this once your migration is no longer blocked due to this issue.

**Comment by Avery-Dunn** [comment]
This issue was addressed in https://github.com/AzureAD/microsoft-authentication-library-for-java/pull/491, and should be fixed as of the recent [1.12.0 release](https://search.maven.org/artifact/com.microsoft.azure/msal4j/1.12.0/jar).

If the issue is still unresolved or if there are any related issues feel free to re-open this thread or open a new one.
